Wrong include directories order

## Description
I've tried to build https://github.com/ros2/rosbag2/pull/687 to check if it works in our environment, but it seems it wont build clean if ros-foxy-rosbag2... packages are installed on the target machine.

I've checked it while building on host machine and reproduced in `ros:foxy` docker container. In both cases `colcon build` finishes with an error:
<details>
    <summary>ERROR</summary>

```
--- stderr: rosbag2_transport_future                                      
In file included from /ws/src/rosbag2_transport/test/rosbag2_transport/rosbag2_transport_test_fixture.hpp:42,
                 from /ws/src/rosbag2_transport/test/rosbag2_transport/record_integration_fixture.hpp:31,
                 from /ws/src/rosbag2_transport/test/rosbag2_transport/test_record_regex.cpp:24:
/ws/src/rosbag2_transport/test/rosbag2_transport/mock_sequential_reader.hpp:28:8: error: ‘void MockSequentialReader::open(const rosbag2_storage::StorageOptions&, const rosbag2_cpp::ConverterOptions&)’ marked ‘override’, but does not override
   28 |   void open(
      |        ^~~~
In file included from /ws/src/rosbag2_transport/test/rosbag2_transport/rosbag2_transport_test_fixture.hpp:43,
                 from /ws/src/rosbag2_transport/test/rosbag2_transport/record_integration_fixture.hpp:31,
                 from /ws/src/rosbag2_transport/test/rosbag2_transport/test_record_regex.cpp:24:
/ws/src/rosbag2_transport/test/rosbag2_transport/mock_sequential_writer.hpp:28:8: error: ‘void MockSequentialWriter::open(const rosbag2_storage::StorageOptions&, const rosbag2_cpp::ConverterOptions&)’ marked ‘override’, but does not override
   28 |   void open(
      |        ^~~~
In file included from /usr/include/c++/9/memory:80,
                 from /opt/ros/foxy/src/gmock_vendor/include/gmock/gmock-actions.h:45,
                 from /opt/ros/foxy/src/gmock_vendor/include/gmock/gmock.h:59,
                 from /ws/src/rosbag2_transport/test/rosbag2_transport/test_record_regex.cpp:15:
/usr/include/c++/9/bits/unique_ptr.h: In instantiation of ‘typename std::_MakeUniq<_Tp>::__single_object std::make_unique(_Args&& ...) [with _Tp = MockSequentialReader; _Args = {}; typename std::_MakeUniq<_Tp>::__single_object = std::unique_ptr<MockSequentialReader, std::default_delete<MockSequentialReader> >]’:
/ws/src/rosbag2_transport/test/rosbag2_transport/rosbag2_transport_test_fixture.hpp:62:90:   required from here
/usr/include/c++/9/bits/unique_ptr.h:857:30: error: invalid new-expression of abstract class type ‘MockSequentialReader’
  857 |     { return unique_ptr<_Tp>(new _Tp(std::forward<_Args>(__args)...)); }
      |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /ws/src/rosbag2_transport/test/rosbag2_transport/rosbag2_transport_test_fixture.hpp:42,
                 from /ws/src/rosbag2_transport/test/rosbag2_transport/record_integration_fixture.hpp:31,
                 from /ws/src/rosbag2_transport/test/rosbag2_transport/test_record_regex.cpp:24:
/ws/src/rosbag2_transport/test/rosbag2_transport/mock_sequential_reader.hpp:25:7: note:   because the following virtual functions are pure within ‘MockSequentialReader’:
   25 | class MockSequentialReader : public rosbag2_cpp::reader_interfaces::BaseReaderInterface
      |       ^~~~~~~~~~~~~~~~~~~~
In file included from /ws/src/rosbag2_transport/test/rosbag2_transport/mock_sequential_reader.hpp:23,
                 from /ws/src/rosbag2_transport/test/rosbag2_transport/rosbag2_transport_test_fixture.hpp:42,
                 from /ws/src/rosbag2_transport/test/rosbag2_transport/record_integration_fixture.hpp:31,
                 from /ws/src/rosbag2_transport/test/rosbag2_transport/test_record_regex.cpp:24:
/opt/ros/foxy/include/rosbag2_cpp/reader_interfaces/base_reader_interface.hpp:40:16: note:  ‘virtual void rosbag2_cpp::reader_interfaces::BaseReaderInterface::open(const rosbag2_cpp::StorageOptions&, const rosbag2_cpp::ConverterOptions&)’
   40 |   virtual void open(
      |                ^~~~
In file included from /usr/include/c++/9/memory:80,
                 from /opt/ros/foxy/src/gmock_vendor/include/gmock/gmock-actions.h:45,
                 from /opt/ros/foxy/src/gmock_vendor/include/gmock/gmock.h:59,
                 from /ws/src/rosbag2_transport/test/rosbag2_transport/test_record_regex.cpp:15:
/usr/include/c++/9/bits/unique_ptr.h: In instantiation of ‘typename std::_MakeUniq<_Tp>::__single_object std::make_unique(_Args&& ...) [with _Tp = MockSequentialWriter; _Args = {}; typename std::_MakeUniq<_Tp>::__single_object = std::unique_ptr<MockSequentialWriter, std::default_delete<MockSequentialWriter> >]’:
/ws/src/rosbag2_transport/test/rosbag2_transport/rosbag2_transport_test_fixture.hpp:63:90:   required from here
/usr/include/c++/9/bits/unique_ptr.h:857:30: error: invalid new-expression of abstract class type ‘MockSequentialWriter’
  857 |     { return unique_ptr<_Tp>(new _Tp(std::forward<_Args>(__args)...)); }
      |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /ws/src/rosbag2_transport/test/rosbag2_transport/rosbag2_transport_test_fixture.hpp:43,
                 from /ws/src/rosbag2_transport/test/rosbag2_transport/record_integration_fixture.hpp:31,
                 from /ws/src/rosbag2_transport/test/rosbag2_transport/test_record_regex.cpp:24:
/ws/src/rosbag2_transport/test/rosbag2_transport/mock_sequential_writer.hpp:25:7: note:   because the following virtual functions are pure within ‘MockSequentialWriter’:
   25 | class MockSequentialWriter : public rosbag2_cpp::writer_interfaces::BaseWriterInterface
      |       ^~~~~~~~~~~~~~~~~~~~
In file included from /ws/src/rosbag2_transport/test/rosbag2_transport/mock_sequential_writer.hpp:23,
                 from /ws/src/rosbag2_transport/test/rosbag2_transport/rosbag2_transport_test_fixture.hpp:43,
                 from /ws/src/rosbag2_transport/test/rosbag2_transport/record_integration_fixture.hpp:31,
                 from /ws/src/rosbag2_transport/test/rosbag2_transport/test_record_regex.cpp:24:
/opt/ros/foxy/include/rosbag2_cpp/writer_interfaces/base_writer_interface.hpp:37:16: note:  ‘virtual void rosbag2_cpp::writer_interfaces::BaseWriterInterface::open(const rosbag2_cpp::StorageOptions&, const rosbag2_cpp::ConverterOptions&)’
   37 |   virtual void open(
      |                ^~~~
make[2]: *** [CMakeFiles/test_record_regex__rmw_fastrtps_cpp.dir/build.make:63: CMakeFiles/test_record_regex__rmw_fastrtps_cpp.dir/test/rosbag2_transport/test_record_regex.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:512: CMakeFiles/test_record_regex__rmw_fastrtps_cpp.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make: *** [Makefile:141: all] Error 2
---
Failed   <<< rosbag2_transport_future [37.1s, exited with code 2]
```
</details>

For some reason `/opt/ros/foxy/include/rosbag2_cpp/writer_interfaces/base_writer_interface.hpp` header was selected  instead of `src/rosbag2_cpp/include/`, so I've build it with exported compile commands and found that [`test_record_regex`](https://github.com/ros2/rosbag2/blob/master/rosbag2_transport/CMakeLists.txt#L155) is compiled like:
```
{
  "directory": "/ws/build/rosbag2_transport_future",
  "command": "/usr/bin/c++  -DDEFAULT_RMW_IMPLEMENTATION=rmw_fastrtps_cpp -DPLUGINLIB__DISABLE_BOOST_FUNCTIONS -DRCUTILS_ENABLE_FAULT_INJECTION -DROS_PACKAGE_NAME=\\\"rosbag2_transport_future\\\" -DSPDLOG_COMPILED_LIB -I/opt/ros/foxy/src/gmock_vendor/include -I/opt/ros/foxy/src/gtest_vendor/include -I/ws/src/rosbag2_transport/include -isystem /ws/install/rosbag2_test_common_future/include -isystem /opt/ros/foxy/include -isystem /ws/install/rosbag2_compression_future/include -isystem /ws/install/rosbag2_cpp_future/include -isystem /ws/install/rosbag2_storage_future/include -isystem /opt/ros/foxy/opt/yaml_cpp_vendor/include -isystem /ws/install/shared_queues_vendor_future/include/moodycamel -isystem /ws/install/shared_queues_vendor_future/include    -Wall -Wextra -Wpedantic -Werror -std=gnu++14 -o CMakeFiles/test_record_regex__rmw_fastrtps_cpp.dir/test/rosbag2_transport/test_record_regex.cpp.o -c /ws/src/rosbag2_transport/test/rosbag2_transport/test_record_regex.cpp",
  "file": "/ws/src/rosbag2_transport/test/rosbag2_transport/test_record_regex.cpp"
}
```
Note than `-isystem /ws/install/rosbag2_cpp_future/include` goes AFTER `-isystem /opt/ros/foxy/include`
While for [`test_record`](https://github.com/ros2/rosbag2/blob/master/rosbag2_transport/CMakeLists.txt#L148) everything seems ok:
```
{
  "directory": "/ws/build/rosbag2_transport_future",
  "command": "/usr/bin/c++  -DDEFAULT_RMW_IMPLEMENTATION=rmw_fastrtps_cpp -DPLUGINLIB__DISABLE_BOOST_FUNCTIONS -DRCUTILS_ENABLE_FAULT_INJECTION -DROS_PACKAGE_NAME=\\\"rosbag2_transport_future\\\" -DSPDLOG_COMPILED_LIB -I/opt/ros/foxy/src/gmock_vendor/include -I/opt/ros/foxy/src/gtest_vendor/include -I/ws/src/rosbag2_transport/src/rosbag2_transport -I/ws/src/rosbag2_transport/include -isystem /ws/install/rosbag2_cpp_future/include -isystem /ws/install/rosbag2_storage_future/include -isystem /ws/install/rosbag2_test_common_future/include -isystem /opt/ros/foxy/include -isystem /opt/ros/foxy/opt/yaml_cpp_vendor/include -isystem /ws/install/rosbag2_compression_future/include -isystem /ws/install/shared_queues_vendor_future/include/moodycamel -isystem /ws/install/shared_queues_vendor_future/include    -Wall -Wextra -Wpedantic -Werror -std=gnu++14 -o CMakeFiles/test_record__rmw_fastrtps_cpp.dir/test/rosbag2_transport/test_record.cpp.o -c /ws/src/rosbag2_transport/test/rosbag2_transport/test_record.cpp",
  "file": "/ws/src/rosbag2_transport/test/rosbag2_transport/test_record.cpp"
}
```
Well, not truly ok, cuz `-isystem /opt/ros/foxy/include` still goes before other includes from workspace. But it compiles at least.

It seems that adding `rosbag2_cpp_future` to [`AMENT_DEPS` for `test_record_regex`](https://github.com/ros2/rosbag2/blob/master/rosbag2_transport/CMakeLists.txt#L158) moves `-isystem /ws/install/rosbag2_cpp_future/include` to position before `-isystem /opt/ros/foxy/include` but I am not sure it is right fix, cuz I dont know how does [`ament_target_dependencies`](https://github.com/ros2/rosbag2/blob/master/rosbag2_transport/cmake/rosbag2_transport_add_gmock.cmake#L60) work and how does it handle transitive dependencies. It might be a problem in ament cmake scripts

Observed behavior is not only for `rosbag2_transport_future` package. It seems that all compile commands for all packages should be reviewed, cuz this behavior might lead to very confusing bugs

### Expected Behavior
Everything compiles successfully and include directories order is right

### Actual Behavior
Error while compiling rosbag2 packages

## To Reproduce
1. Run ros container with rosbag2 inside:
```
docker run --rm -it ros:foxy /bin/bash
```
2. Install deps:
```
# Install deps
apt update && apt install ros-foxy-test-msgs pybind11-dev
```
3. Create workspace and clone rosbag2: 
```
mkdir -p ws/src
git clone https://github.com/ros2/rosbag2.git -b emersonknapp/futurize-foxy ws/src
cd ws
```
4. Build:
```
colcon build --ament-cmake-args -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
```
5. Check for compile commands: `cat build/rosbag2_transport_future/compile_commands.json`

## System (please complete the following information)
 - OS: Ubuntu Bionic
 - ROS 2 Distro: foxy
 - Version: Reproduced in emersonknapp/futurize-foxy, but master branch still affected

--- a/rosbag2_compression/CMakeLists.txt
+++ b/rosbag2_compression/CMakeLists.txt
@@ -75,6 +75,8 @@ if(BUILD_TESTING)
     test/rosbag2_compression/fake_compressor.cpp
     test/rosbag2_compression/fake_decompressor.cpp)
   target_link_libraries(fake_plugin ${PROJECT_NAME})
+  # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+  ament_target_dependencies(fake_plugin rosbag2_cpp rosbag2_storage)
   install(
     TARGETS fake_plugin
     ARCHIVE DESTINATION lib
@@ -86,22 +88,28 @@ if(BUILD_TESTING)
   ament_add_gmock(test_compression_factory
     test/rosbag2_compression/test_compression_factory.cpp)
   target_link_libraries(test_compression_factory ${PROJECT_NAME})
+  # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+  ament_target_dependencies(test_compression_factory rosbag2_cpp rosbag2_storage)
 
   ament_add_gmock(test_compression_options
     test/rosbag2_compression/test_compression_options.cpp)
   target_include_directories(test_compression_options PUBLIC include)
   target_link_libraries(test_compression_options ${PROJECT_NAME})
+  # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+  ament_target_dependencies(test_compression_options rosbag2_cpp rosbag2_storage)
 
   ament_add_gmock(test_sequential_compression_reader
     test/rosbag2_compression/test_sequential_compression_reader.cpp)
   target_include_directories(test_sequential_compression_reader PUBLIC include)
   target_link_libraries(test_sequential_compression_reader ${PROJECT_NAME})
+  # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
   ament_target_dependencies(test_sequential_compression_reader rosbag2_cpp rosbag2_storage)
 
   ament_add_gmock(test_sequential_compression_writer
     test/rosbag2_compression/test_sequential_compression_writer.cpp)
   target_include_directories(test_sequential_compression_writer PUBLIC include)
   target_link_libraries(test_sequential_compression_writer ${PROJECT_NAME})
+  # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
   ament_target_dependencies(test_sequential_compression_writer rosbag2_cpp rosbag2_storage)
 endif()
 

--- a/rosbag2_compression_zstd/CMakeLists.txt
+++ b/rosbag2_compression_zstd/CMakeLists.txt
@@ -38,7 +38,10 @@ target_include_directories(${PROJECT_NAME}
 ament_target_dependencies(${PROJECT_NAME}
   rcpputils
   rosbag2_compression
-  zstd)
+  zstd
+  # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+  rosbag2_cpp
+  rosbag2_storage)
 target_compile_definitions(${PROJECT_NAME} PRIVATE ROSBAG2_COMPRESSION_ZSTD_BUILDING_DLL)
 pluginlib_export_plugin_description_file(rosbag2_compression plugin_description.xml)
 
@@ -69,9 +72,11 @@ if(BUILD_TESTING)
 
   ament_add_gmock(test_zstd_compressor
     test/rosbag2_compression_zstd/test_zstd_compressor.cpp)
-  target_include_directories(test_zstd_compressor PUBLIC include)
   target_link_libraries(test_zstd_compressor ${PROJECT_NAME})
-  ament_target_dependencies(test_zstd_compressor rclcpp rosbag2_test_common)
+
+  ament_target_dependencies(test_zstd_compressor rclcpp rosbag2_test_common
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+    rosbag2_cpp rosbag2_storage zstd rosbag2_compression)
 endif()
 
 ament_package()

--- a/rosbag2_cpp/CMakeLists.txt
+++ b/rosbag2_cpp/CMakeLists.txt
@@ -140,8 +140,9 @@ if(BUILD_TESTING)
   ament_add_gmock(test_converter_factory
     test/rosbag2_cpp/test_converter_factory.cpp)
   if(TARGET test_converter_factory)
-    target_include_directories(test_converter_factory PRIVATE include)
     target_link_libraries(test_converter_factory ${PROJECT_NAME})
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+    ament_target_dependencies(test_converter_factory rosbag2_storage)
   endif()
 
   ament_add_gmock(test_typesupport_helpers
@@ -149,6 +150,8 @@ if(BUILD_TESTING)
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
   if(TARGET test_typesupport_helpers)
     target_link_libraries(test_typesupport_helpers ${PROJECT_NAME})
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+    ament_target_dependencies(test_typesupport_helpers rosbag2_storage rosbag2_test_common)
   endif()
 
   ament_add_gmock(test_info
@@ -156,27 +159,33 @@ if(BUILD_TESTING)
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
   if(TARGET test_info)
     target_link_libraries(test_info ${PROJECT_NAME})
-    ament_target_dependencies(test_info rosbag2_test_common)
+    ament_target_dependencies(test_info rosbag2_test_common
+      # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+      rosbag2_storage)
   endif()
 
   ament_add_gmock(test_sequential_reader
     test/rosbag2_cpp/test_sequential_reader.cpp)
   if(TARGET test_sequential_reader)
-    ament_target_dependencies(test_sequential_reader rosbag2_storage)
     target_link_libraries(test_sequential_reader ${PROJECT_NAME})
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+    ament_target_dependencies(test_sequential_reader rosbag2_storage)
   endif()
 
   ament_add_gmock(test_storage_without_metadata_file
     test/rosbag2_cpp/test_storage_without_metadata_file.cpp)
   if(TARGET test_storage_without_metadata_file)
-    ament_target_dependencies(test_storage_without_metadata_file rosbag2_storage)
     target_link_libraries(test_storage_without_metadata_file ${PROJECT_NAME})
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+    ament_target_dependencies(test_storage_without_metadata_file rosbag2_storage)
   endif()
 
   ament_add_gmock(test_message_cache
     test/rosbag2_cpp/test_message_cache.cpp)
   if(TARGET test_message_cache)
     target_link_libraries(test_message_cache ${PROJECT_NAME})
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+    ament_target_dependencies(test_message_cache rosbag2_storage)
   endif()
 
   # If compiling with gcc, run this test with sanitizers enabled
@@ -207,15 +216,17 @@ if(BUILD_TESTING)
   ament_add_gmock(test_sequential_writer
     test/rosbag2_cpp/test_sequential_writer.cpp)
   if(TARGET test_sequential_writer)
-    ament_target_dependencies(test_sequential_writer rosbag2_storage)
     target_link_libraries(test_sequential_writer ${PROJECT_NAME})
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+    ament_target_dependencies(test_sequential_writer rosbag2_storage)
   endif()
 
   ament_add_gmock(test_multifile_reader
     test/rosbag2_cpp/test_multifile_reader.cpp)
   if(TARGET test_multifile_reader)
-    ament_target_dependencies(test_multifile_reader rosbag2_storage)
     target_link_libraries(test_multifile_reader ${PROJECT_NAME})
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+    ament_target_dependencies(test_multifile_reader rosbag2_storage)
   endif()
 endif()
 

--- a/rosbag2_tests/CMakeLists.txt
+++ b/rosbag2_tests/CMakeLists.txt
@@ -40,19 +40,23 @@ if(BUILD_TESTING)
   find_package(rosbag2_test_common REQUIRED)
   find_package(std_msgs REQUIRED)
   find_package(test_msgs REQUIRED)
+  find_package(zstd_vendor REQUIRED)
 
   ament_add_gmock(test_rosbag2_record_end_to_end
     test/rosbag2_tests/test_rosbag2_record_end_to_end.cpp
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
   if(TARGET test_rosbag2_record_end_to_end)
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
     ament_target_dependencies(test_rosbag2_record_end_to_end
       rclcpp
+      rosbag2_cpp
       rosbag2_compression
       rosbag2_compression_zstd
       rosbag2_storage
       rosbag2_storage_default_plugins
       rosbag2_test_common
-      test_msgs)
+      test_msgs
+      zstd_vendor)
     ament_add_test_label(test_rosbag2_record_end_to_end xfail)
   endif()
 
@@ -60,6 +64,7 @@ if(BUILD_TESTING)
     test/rosbag2_tests/test_rosbag2_play_end_to_end.cpp
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
   if(TARGET test_rosbag2_play_end_to_end)
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
     ament_target_dependencies(test_rosbag2_play_end_to_end
       rclcpp
       rosbag2_storage
@@ -73,6 +78,7 @@ if(BUILD_TESTING)
     test/rosbag2_tests/test_rosbag2_info_end_to_end.cpp
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
   if(TARGET test_rosbag2_info_end_to_end)
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
     ament_target_dependencies(test_rosbag2_info_end_to_end
       rosbag2_storage
       rosbag2_test_common)
@@ -83,16 +89,19 @@ if(BUILD_TESTING)
     test/rosbag2_tests/test_converter.cpp
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
   if(TARGET test_converter)
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
     ament_target_dependencies(test_converter
       rosbag2_cpp
       rosbag2_test_common
-      test_msgs)
+      test_msgs
+      rosbag2_storage)
   endif()
 
   ament_add_gmock(test_rosbag2_cpp_api
     test/rosbag2_tests/test_rosbag2_cpp_api.cpp
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
   if(TARGET test_rosbag2_cpp_api)
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
     ament_target_dependencies(test_rosbag2_cpp_api
       rclcpp
       rosbag2_cpp

--- a/rosbag2_transport/CMakeLists.txt
+++ b/rosbag2_transport/CMakeLists.txt
@@ -68,6 +68,7 @@ ament_target_dependencies(rosbag2_transport_py
   rosbag2_cpp
   rosbag2_storage
   rmw
+  shared_queues_vendor
 )
 # Causes the visibility macros to use dllexport rather than dllimport,
 # which is appropriate when building the dll but not consuming it.
@@ -94,17 +95,20 @@ function(create_tests_for_rmw_implementation)
   rosbag2_transport_add_gmock(test_record_all
     test/rosbag2_transport/test_record_all.cpp
     LINK_LIBS rosbag2_transport
-    AMENT_DEPS rosbag2_cpp rosbag2_storage test_msgs rosbag2_test_common)
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+    AMENT_DEPS rosbag2_cpp rosbag2_storage test_msgs rosbag2_test_common rosbag2_compression shared_queues_vendor)
 
   rosbag2_transport_add_gmock(test_record_all_no_discovery
     test/rosbag2_transport/test_record_all_no_discovery.cpp
     LINK_LIBS rosbag2_transport
-    AMENT_DEPS rosbag2_cpp rosbag2_storage test_msgs rosbag2_test_common)
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+    AMENT_DEPS rosbag2_cpp rosbag2_storage test_msgs rosbag2_test_common rosbag2_compression shared_queues_vendor)
 
   rosbag2_transport_add_gmock(test_play_timing
     test/rosbag2_transport/test_play_timing.cpp
     LINK_LIBS rosbag2_transport
-    AMENT_DEPS rosbag2_cpp rosbag2_storage test_msgs rosbag2_test_common)
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+    AMENT_DEPS rosbag2_cpp rosbag2_storage test_msgs rosbag2_test_common rosbag2_compression shared_queues_vendor)
 
   rosbag2_transport_add_gmock(test_rosbag2_node
     src/rosbag2_transport/generic_publisher.cpp
@@ -118,6 +122,7 @@ function(create_tests_for_rmw_implementation)
       $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
       $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/rosbag2_transport>
       $<INSTALL_INTERFACE:include>
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
     AMENT_DEPS
       ament_index_cpp
       rclcpp
@@ -125,7 +130,9 @@ function(create_tests_for_rmw_implementation)
       rosbag2_storage
       rosbag2_test_common
       test_msgs
-      yaml_cpp_vendor)
+      yaml_cpp_vendor
+      rosbag2_compression
+      shared_queues_vendor)
 
   rosbag2_transport_add_gmock(test_qos
     src/rosbag2_transport/qos.cpp
@@ -133,6 +140,7 @@ function(create_tests_for_rmw_implementation)
     INCLUDE_DIRS
       $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
       $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/rosbag2_transport>
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
     AMENT_DEPS
       rclcpp
       rosbag2_test_common
@@ -147,7 +155,8 @@ function(create_tests_for_rmw_implementation)
 
   rosbag2_transport_add_gmock(test_record
     test/rosbag2_transport/test_record.cpp
-    AMENT_DEPS rosbag2_cpp rosbag2_storage test_msgs rosbag2_test_common
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+    AMENT_DEPS rosbag2_cpp rosbag2_storage test_msgs rosbag2_test_common rosbag2_compression shared_queues_vendor
     INCLUDE_DIRS $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/rosbag2_transport>
     LINK_LIBS rosbag2_transport
     ${SKIP_TEST})
@@ -155,7 +164,8 @@ function(create_tests_for_rmw_implementation)
   rosbag2_transport_add_gmock(test_record_regex
     test/rosbag2_transport/test_record_regex.cpp
     LINK_LIBS rosbag2_transport
-    AMENT_DEPS test_msgs rosbag2_test_common
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+    AMENT_DEPS test_msgs rosbag2_test_common rosbag2_compression shared_queues_vendor rosbag2_cpp rosbag2_storage
     ${SKIP_TEST})
 
   rosbag2_transport_add_gmock(test_play
@@ -163,20 +173,23 @@ function(create_tests_for_rmw_implementation)
     test/rosbag2_transport/test_play.cpp
     INCLUDE_DIRS $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/rosbag2_transport>
     LINK_LIBS rosbag2_transport
-    AMENT_DEPS rosbag2_cpp rosbag2_storage test_msgs rosbag2_test_common
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+    AMENT_DEPS rosbag2_cpp rosbag2_storage test_msgs rosbag2_test_common rosbag2_compression shared_queues_vendor
     ${SKIP_TEST})
 
   rosbag2_transport_add_gmock(test_play_loop
     test/rosbag2_transport/test_play_loop.cpp
     ${SKIP_TEST}
     LINK_LIBS rosbag2_transport
-    AMENT_DEPS rosbag2_cpp rosbag2_storage test_msgs rosbag2_test_common)
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+    AMENT_DEPS rosbag2_cpp rosbag2_storage test_msgs rosbag2_test_common rosbag2_compression shared_queues_vendor)
 
   rosbag2_transport_add_gmock(test_play_topic_remap
     test/rosbag2_transport/test_play_topic_remap.cpp
     ${SKIP_TEST}
     LINK_LIBS rosbag2_transport
-    AMENT_DEPS rosbag2_cpp rosbag2_storage test_msgs rosbag2_test_common)
+    # Needs to be removed/revised after https://github.com/ros2/ros2/issues/1150 gets fixed
+    AMENT_DEPS rosbag2_cpp rosbag2_storage test_msgs rosbag2_test_common rosbag2_compression shared_queues_vendor)
 endfunction()
 
 if(BUILD_TESTING)

